### PR TITLE
added components and api

### PR DESCRIPTION
**Pull request by pranav96vaidya**

### CODE REVIEW

**Comment by niteshthakur5** [comment]
Remove src/.DS_Store file and add it to .gitignore.

File: src/app/dashboard/dashboard.component.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  getState: Observable<any>;
+  isAuthenticated: boolean = false;
+  fetchDone = false;
+  users: string;
+  employeeList: string;
+  month = new Date().getMonth();
+  months: string[] = ["January", "February", "March", "April", "May",
+    "June", "July", "August", "September", "October", "November", "December"];
+  currentMonth = this.months[this.month];
+  years: number[] = [2017, 2018, 2019];
```
Comment (shikhashakarwar) [inline src/app/dashboard/dashboard.component.ts]: `years` array should hold range from prev to current running year. If will fail if the year will 2020. Please, those cases also.

File: src/app/dashboard/dashboard.component.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    // this.employeeList = JSON.parse(localStorage.getItem("employee"));
+    // this.users = this.employeeList['data'];
+    // console.log(this.currentMonth);
+    this.employeeForm = new FormGroup({
+      'emp': new FormControl('', Validators.required),
+      'yearVal': new FormControl(this.currentYear),
+      'monthVal': new FormControl(this.currentMonth)
+    })
+
+    // this.employeeForm.controls['emp'].valueChanges.subscribe((value) => {
```
Comment (shikhashakarwar) [inline src/app/dashboard/dashboard.component.ts]: remove comment code if it not required.

File: src/app/drag-drop.directive.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import { Directive, Output, Input, EventEmitter, HostBinding, HostListener } from '@angular/core';
+
+@Directive({
+  selector: '[appDragDrop]'
+})
+export class DragDropDirective {
+  @Output() onFileDropped = new EventEmitter<any>();
+
+  @HostBinding('style.background-color') private background = '#f5fcff'
+  @HostBinding('style.opacity') private opacity = '1'
```
Comment (shikhashakarwar) [inline src/app/drag-drop.directive.ts]: Semicolon is missing at too many places.

File: src/app/employee-salary/employee-salary.component.html [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  <div class="select-file-error">{{errorMsg}}</div>
+  <div class="navigate-desc">To navigate to home page <a routerLink="/home">Click here</a></div>
+</ng-container>
+<div class="row btn-container" *ngIf="fetchDone && !errorMsg">
+  <div class="col-3">
+    <button class="btn btn-primary back-btn" (click)="previousPage()">Back</button>
+  </div>
+  <div class="col-6 head-content">Salary Slip of Employees</div>
+  <div class="col-3 send-btn-container">
+    <button class="btn btn-primary delete-btn" [disabled]="checkedList.length == 0" (click)="deleteItem()">Delete</button>
```
Comment (shikhashakarwar) [inline src/app/employee-salary/employee-salary.component.html]: We can simply add a check on `checkedList && checkedList.length`. Don't need to check length ==0

File: src/app/employee-salary/employee-salary.component.html [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    <button class="btn btn-primary send-btn" [disabled]="checkedList.length == 0" (click)="sendMail()">Send Mail</button>
```
Comment (shikhashakarwar) [inline src/app/employee-salary/employee-salary.component.html]: same here

File: src/app/salary-list/salary-list.component.ts [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    console.log(this.route);
+    this.route.params.subscribe(data => {
+      this.empId = data['empId'];
+      console.log(data);
+    });
+
+    this.empDetails.getEmpdetail(this.empId)
+    .subscribe(response => {
+      console.log(response);
+      if(response['data'] == null) {
```
Comment (shikhashakarwar) [inline src/app/salary-list/salary-list.component.ts]: check this to `response['data']`

**Pull request merged**